batcher/orderbook.mligo

```diff
@@ -24,13 +25,89 @@ type match_calculation_type = EXACT_MATCH | LEFT_PARTIAL | RIGHT_PARTIAL
 
 let empty () : t = {bids = ([] : order list); asks = ([] : order list)}
 
+let constant_number = Constants.constant_number
+
+(* 
+  The actual price when the sides aren't the same.
+  | Inversion   | P-10bps          | P         | P+10bps          |
+  |-------------|:----------------:|----------:|-----------------:|
+  | *True*      | P * 1.0001       | P         | P / 1.0001       |
+*)
+let compute_true_inversion_equivalent_token 
+  (order : order) 
+  (float_amount : Float.t) 
+  (exchange_rate : exchange_rate) 
+  (tolerance : tolerance) : nat = 
+    if order.swap.from.token = exchange_rate.swap.from.token then 
+      match tolerance with 
+      | MINUS -> 
+        let equivalent_amount = Float.mul (Float.mul float_amount exchange_rate.rate) constant_number in 
+        Math.get_rounded_number equivalent_amount
+      | EXACT -> 
+        let equivalent_amount = Float.mul float_amount exchange_rate.rate in 
+        Math.get_rounded_number equivalent_amount
+      | PLUS ->
+        let equivalent_amount = Float.div (Float.mul float_amount exchange_rate.rate) constant_number in 
+        Math.get_rounded_number equivalent_amount
+    else 
+      match tolerance with 
+      | MINUS -> 
+        let equivalent_amount = Float.div float_amount (Float.mul exchange_rate.rate constant_number) in 
+        Math.get_rounded_number equivalent_amount
+      | EXACT -> 
+        let equivalent_amount = Float.div float_amount exchange_rate.rate in 
+        Math.get_rounded_number equivalent_amount
+      | PLUS ->
+        let equivalent_amount = Float.div (Float.mul float_amount constant_number) exchange_rate.rate in 
+        Math.get_rounded_number equivalent_amount
+
+
+(* 
+  The actual price when the sides are the same.
```

Comment:
I don't understand what you mean by "when the sides are the same"?

Reply:
For example, the current exchange rate is `tzBTC/USDT` and the buy side is `tzBTC`.

Reply:
Ah okay, I see what you are saying now.